### ISSUE TITLE
Emptying a theme's welcome message throws an error.

### ISSUE THREAD

**Issue by Hexmage** [magento/magento2#37748]
### Preconditions and environment

- Magento 2.4.6-p1
- Any theme that shows the default welcome message. (luma or blank will do)

### Steps to reproduce

1. In the Adminpanel navigate to Content -> Design -> Configuration.
2. Select your default storeview.
3. Go to Other Settings -> Header -> Welcome text, and empty it.
4. Flush cache.
5. Open the frontend homepage.

### Expected result

No Welcome message is shown when you visit the homepage.

### Actual result

```
TypeError: addslashes(): Argument #1 ($string) must be of type string, null given in ../vendor/magento/framework/Escaper.php:440
Stack trace:
#0 ..vendor/magento/framework/Escaper.php(440): addslashes()
#1 ..vendor/magento/module-theme/Block/Html/Header.php(62): Magento\Framework\Escaper->escapeQuote()
#2 ..vendor/magento/module-theme/view/frontend/templates/html/header.phtml(11): Magento\Theme\Block\Html\Header->getWelcome()
#3 ..vendor/magento/framework/View/TemplateEngine/Php.php(71): include('...')
#4 ..vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\TemplateEngine\Php->render()
#5 ..vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent()
#6 ..vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\TemplateEngine\Php\Interceptor->Magento\Framework\Interception\{closure}()
#7 ..generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php(23): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callPlugins()
#8 ..vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php\Interceptor->render()
#9 ..vendor/magento/framework/View/Element/Template.php(293): Magento\Framework\View\Element\Template->fetchView()
#10 ..vendor/magento/framework/View/Element/AbstractBlock.php(1117): Magento\Framework\View\Element\Template->_toHtml()
#11 ..vendor/magento/framework/View/Element/AbstractBlock.php(1121): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#12 ..vendor/magento/framework/View/Element/AbstractBlock.php(670): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#13 ..vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#14 ..vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock()
#15 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#16 ..vendor/magento/framework/View/Element/Html/Links.php(63): Magento\Framework\View\Layout->renderElement()
#17 ..vendor/magento/framework/View/Element/Html/Links.php(83): Magento\Framework\View\Element\Html\Links->renderLink()
#18 ..vendor/magento/framework/View/Element/AbstractBlock.php(1117): Magento\Framework\View\Element\Html\Links->_toHtml()
#19 ..vendor/magento/framework/View/Element/AbstractBlock.php(1121): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#20 ..vendor/magento/framework/View/Element/AbstractBlock.php(670): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#21 ..vendor/magento/framework/View/Layout.php(578): Magento\Framework\View\Element\AbstractBlock->toHtml()
#22 ..vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock()
#23 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#24 ..vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#25 ..vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#26 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#27 ..vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#28 ..vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#29 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#30 ..vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#31 ..vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#32 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#33 ..vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#34 ..vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#35 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#36 ..vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout->renderElement()
#37 ..vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer()
#38 ..vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout->renderNonCachedElement()
#39 ..vendor/magento/framework/View/Layout.php(975): Magento\Framework\View\Layout->renderElement()
#40 ..vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Layout->getOutput()
#41 ..vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Layout\Interceptor->___callParent()
#42 ..vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Layout\Interceptor->Magento\Framework\Interception\{closure}()
#43 ..generated/code/Magento/Framework/View/Layout/Interceptor.php(41): Magento\Framework\View\Layout\Interceptor->___callPlugins()
#44 ..vendor/magento/framework/View/Result/Page.php(260): Magento\Framework\View\Layout\Interceptor->getOutput()
#45 ..vendor/magento/framework/View/Result/Layout.php(171): Magento\Framework\View\Result\Page->render()
#46 ..vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\Result\Layout->renderResult()
#47 ..vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\Result\Page\Interceptor->___callParent()
#48 ..vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\Result\Page\Interceptor->Magento\Framework\Interception\{closure}()
#49 ..generated/code/Magento/Framework/View/Result/Page/Interceptor.php(32): Magento\Framework\View\Result\Page\Interceptor->___callPlugins()
#50 ..vendor/magento/framework/App/Http.php(120): Magento\Framework\View\Result\Page\Interceptor->renderResult()
#51 ..vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http->launch()
#52 ..pub/index.php(50): Magento\Framework\App\Bootstrap->run()
```

### Additional information

Looks like strict typing is throwing an error here.

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by koushikch7** [comment]
@magento give me 2.4-6-p1 instance

**Comment by engcom-Bravo** [comment]
Hi @Hexmage,

Thank you for reporting and collaboration.

Verified the issue on Magento 2.4-develop instance and the issue is not reproducible.Kindly refer the screenshots.

**Steps to reproduce**

- In the Adminpanel navigate to Content -> Design -> Configuration.
- Select your default storeview.
- Go to Other Settings -> Header -> Welcome text, and empty it.
- Flush cache.
- Open the frontend homepage.



<img width="1444" alt="Screenshot 2023-07-12 at 10 37 59 AM" src="https://github.com/magento/magento2/assets/51680745/c3812dc9-577c-43dd-bddf-6d66e48ba41f">



<img width="1389" alt="Screenshot 2023-07-12 at 10 38 05 AM" src="https://github.com/magento/magento2/assets/51680745/5ea25ecd-8f95-4fc8-8bda-2eed0e40adde">


There is no welcome message while visiting the frontend for the default store view.


Kindly recheck the behaviour on Magento 2.4-develop instance and elaborate steps to reproduce if the issue is still reproducible.


Thanks.

**Comment by Hexmage** [comment]
@engcom-Bravo 
What php version does the Magento instance get spinned up on? Because this should only be happening in php 8.2.

**Comment by dmanners** [comment]
We are having the same issue today on a Magento cloud instance running php 8.1.x

**Comment by Hexmage** [comment]
@dmanners 
A bit hacky but that does fix the issue.

` $this->_data['welcome'] =  $this->escaper->escapeQuote($this->_data['welcome'] ?? '', true);`

Think I would have fixed it like this.

**Comment by dmanners** [comment]
@Hexmage yeah that looks like a nice clean solution. My main thought was if I already know it is an empty string do I really need/want to run it through the escaper and the translation class.

**Comment by engcom-Bravo** [comment]
Hi @Hexmage,

Thanks for your update.

Verified the issue on Magento 2.4-develop instance with php 8.2 and the issue is not reproducible.Kindly refer the screenshots.

**Steps to reproduce**

- In the Adminpanel navigate to Content -> Design -> Configuration.
- Select your default storeview.
- Go to Other Settings -> Header -> Welcome text, and empty it.
- Flush cache.
- Open the frontend homepage.


<img width="668" alt="Screenshot 2023-07-13 at 1 57 42 PM" src="https://github.com/magento/magento2/assets/51680745/61810909-e911-4139-8f38-7b02620f7c9d">


<img width="1483" alt="Screenshot 2023-07-13 at 1 57 26 PM" src="https://github.com/magento/magento2/assets/51680745/e7a4d62c-18c7-4cf3-abb6-4b5a826b1a35">






<img width="1473" alt="Screenshot 2023-07-13 at 1 56 01 PM" src="https://github.com/magento/magento2/assets/51680745/127b8664-da1d-4b17-9b95-1aa0ff802078">



There is no welcome message while visiting the frontend for the default store view.Please let us know still if we are missing.


Thanks.

**Comment by hostep** [comment]
This sounds like a duplicate of https://github.com/magento/magento2/issues/37208

Fix might be this one: https://github.com/magento/magento2/issues/37208#issuecomment-1481743137 and is already included in Magento 2.4.7-beta1

**Comment by engcom-Bravo** [comment]
@hostep Thanks for your contribution & collaboration over here.

The issue is fixed with this commit 4b5c1dc and it is no longer reproducible in Magento 2.4-develop instance and also changes included in Magento latest version.we have verified the issue in Magento 2.4.7-beta 1 and it is not reproducible.


Thanks.

**Comment by Hexmage** [comment]
Must have missed that issue when I searched for "welcome" (probably because I only searched through open issues). I'll close this now as it looks like its already fixed.
